src/sidebar/test/local-storage-test.js

```diff
@@ -1,15 +1,35 @@
 'use strict';
 
 var angular = require('angular');
+var service = require('../local-storage');
 
 describe('sidebar.localStorage', () => {
   var fakeWindow;
 
   before(() =>
     angular.module('h', [])
-      .service('localStorage', require('../local-storage'))
+      .service('localStorage', service)
   );
 
+  context('when accessing localStorage throws an Error', () => {
+    it('returns the fallback implementation', () => {
+      var badWindow = {};
+      var fakeWindow = {};
+
+      Object.defineProperty(badWindow, 'localStorage', {
+        get: () => {
+          throw Error('denied');
+        },
+      });
+
+      var prototypes = [badWindow, fakeWindow]
+          .map(service)
+          .map(Object.getPrototypeOf)
+      ;
+      assert.strictEqual(prototypes[0], prototypes[1]);
```

Comment:
I think the prototype check here will return true regardless of the implementation because the service constructor returns an object literal rather than an instance of a class.

The way I would suggest testing this is to wrap the context block below which tests for behaviour when localStorage methods throw in a loop which iterates over different unusable localStorage implementations. eg:

```js
[{
  window: /* window with localStorage getter that throws */,
},{
  window: /* window with localStorage methods that throw */,
}].forEach((win) => {
  context('when browser localStorage is *not* accessible', () => {
    ...
  });
});
```

I'm happy to do this if you're busy. Let me know either way.